File: Sem1-2/CPP_Java2024/List3/JAVA/Square.java
public class Square extends Quad
{
    protected final double side;

    public Square(double side) throws IllegalArgumentException
    {
        super("Square");
        if(side < 0)
        {
            throw new IllegalArgumentException("Square side cannot be negative, got: " + side);
        }
        this.side = side;
    }

    @Override
    public double Circumference()
    {
        return 4.0 * side;
    }

    @Override
    public double Area()
    {
        return side * side;
    }
}

File: Sem1-2/CPP_Java2024/List3/CPP/src/Square.cpp
#include <Square.hpp>
#include <utils.hpp>

#include <stdexcept>

Square::Square(double side) : Quad("Square"), side(side)
{
    if(side < 0)
    {
        throw std::invalid_argument(LOG_LOC() + "Square side cannot be negative, got: " + std::to_string(side));
    }
}

double Square::Circumference() const
{
    return 4.0 * side;
}

double Square::Area() const
{
    return side * side;
}